### DISCUSSION TITLE
Nuxt 3/VUE does not start anymore after adding supabase. Warnings on stream module and "global" not defined

### DISCUSSION THREAD

**Discussion by tinadanweb** [supabase/supabase#16737] [Questions]
![Screenshot 2023-08-23 at 13 25 16](https://github.com/supabase/supabase/assets/85861795/15410639-ed4b-4d4d-ac17-3f4538608321)

Module "stream" has been externalized for browser compatibility. Cannot access "stream.PassThrough" in client code. See http://vitejs.dev/guide/troubleshooting.html#module-externalized-for-browser-compatibility for more details. 

**My environment is:** 
package.json
`
{
  "name": "nuxt-app",
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "devDependencies": {
    "@nuxt/devtools": "latest",
    "@nuxtjs/supabase": "^1.0.2",
    "@nuxtjs/tailwindcss": "^6.8.0",
    "@types/node": "^18.17.3",
    "nuxt": "^3.6.5"
  },
  "dependencies": {
    "@nuxtjs/color-mode": "^3.3.0",
    "vite": "^4.4.9",
    "webpack": "^5.88.2"
  }
}
`

my NUXT 3 set-up 
`
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: [
    '@nuxtjs/tailwindcss',
    '@nuxtjs/color-mode',
    '@nuxtjs/supabase',
  ],
  colorMode: {
    classSuffix: ''
  }
})
`

after the warnings i get also this error of 

 I've tried adding `vite: { global: {}}` as per some stuckoverflow ideas ... did not work. Installed vite as a dependency too.... 
nothing works 😭   I am 🙉 

If I take out supabase from the project, no errors everything works fine again.  Very weird. I've clean cache, yarn cache reinstalled yarn - nothing works. I'm on a mac M1 environment. 

Any ideas? 🙈

**Comment by aladin-bilalagic** [comment]
I got the same issue with React and Vite. Also tried pretty much everything I could find related to this but couldn't get it to work..

**Comment by aladin-bilalagic** [comment]
I found a way to fix it.
Polyfill for stream was needed.

**Steps to fix it:**
1. npm i stream-browserify
2. add `stream: "stream-browserify"` inside vite config alias, example below:

```javascript
export default defineConfig({
  plugins: [react()],
  resolve: {
    alias: {
      "@": path.resolve(__dirname, "./src"),
      stream: "stream-browserify",
    },
  },
});
```

3. add
 ```javascript
<script>
  if (global === undefined) {
    var global = window;
  }
</script>
```

inside head tag in index.html

That got rid of the error and now works for me.
